### DISCUSSION TITLE
is onlook self-hostable?

### DISCUSSION THREAD

**Discussion by t3rry327** [onlook-dev/onlook#863] [Questions] [answered]
As this is made in typescript (i know with electron but i guess its not too relevant), is there any way of self hosting it?

**Comment by Kitenite** [comment]
@t3rry327 , yes, you can build and package the app locally and use it without dependencies. 

```
bun run build && cd apps/studio && bun run package
```
https://github.com/onlook-dev/onlook/blob/0074d5ce32b2edfe4a2546735c683eb8548890e7/apps/studio/package.json#L24-L28

If you'd like to set up everything, including AI, you can just copy the .env.example into a .env file and fill in all the keys. Does this help? 
I can create a doc for this if you run into issues. None of the keys are required to run Onlook but they progressively add more features.

https://github.com/onlook-dev/onlook/blob/5042b68030d01070c2b7eef4c8aec3b6b2fddf06/apps/studio/.env.example#L1

**Reply by t3rry327** [reply]
sorry if i didnt explain myself correctly, i mean WEB-host it

**Reply by Kitenite** [reply]
Unfortunately, This is not currently doable unless you know of a way of exposing Electron app's front-end

**Reply by Kitenite** [reply]
What is your usecase here in case we have a better workaround?

**Reply by t3rry327** [reply]
just wanted to self host a web version using the server hardware and just using the browser to work, but i understand the limitations

**Reply by Kitenite** [reply]
I'm not aware of a way to do this unfortunately :(

**Reply by nazim121** [reply]
Unfortunately, This is not currently doable unless you know of a way of exposing. [https://unicyclehero.net/](url)

**Reply by Kitenite** [reply, marked as answer]
Well actualy, since we have the web version, you can now do that @t3rry327 
https://docs.onlook.com/docs/developer/running-locally

**Comment by lx1054331851** [comment]
> @t3rry327 , yes, you can build and package the app locally and use it without dependencies. 
> 
> 
> 
> ```
> 
> bun run build && cd apps/studio && bun run package
> 
> ```
> 
> https://github.com/onlook-dev/onlook/blob/0074d5ce32b2edfe4a2546735c683eb8548890e7/apps/studio/package.json#L24-L28
> 
> 
> 
> If you'd like to set up everything, including AI, you can just copy the .env.example into a .env file and fill in all the keys. Does this help? 
> 
> I can create a doc for this if you run into issues. None of the keys are required to run Onlook but they progressively add more features.
> 
> 
> 
> https://github.com/onlook-dev/onlook/blob/5042b68030d01070c2b7eef4c8aec3b6b2fddf06/apps/studio/.env.example#L1
> 
> 
> 
> 

Can I use azure open ai api？
I look through the source code,but I can't find the llm process logical code.
Can you tell me which file contains the relation code?
Thank you so much!

**Reply by Kitenite** [reply]
Hey @lx1054331851 sorry I don't get notified for these comments. You can add new providers below. I would encourage you to open a PR if you do :) 
https://github.com/onlook-dev/onlook/blob/main/apps/studio/electron/main/chat/llmProvider.ts#L20-L31

**Comment by nazim121** [comment]
just wanted to self host a web version using the server hardware and just using the browser to work, but i understand the limitations
